### ISSUE TITLE
Fixing "property has no initializer and is not definitely assigned in the constructor" error causes incorrect member-access and whitespace lint errors

### ISSUE THREAD

**Issue by ramanujam-raman** [palantir/tslint#3716]
### Bug Report

- __TSLint version__:5.9.1
- __TypeScript version__:2.7.1
- __Running TSLint via__: (pick one) Visual Studio Code

#### TypeScript code being linted

```ts
// code snippet
export abstract class Foo {
  public name!: string;
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended"
  ],
}
```

#### Actual behavior
member-access and whitespace lint errors
#### Expected behavior
no lint errors

**Comment by reflectiondm** [comment]
@ramanujam-raman have you tried to restart VSCode after updating? It seem to have fixed the problem for  me.

**Comment by ramanujam-raman** [comment]
Yes, I have restarted it since. I think the issue is really lack of support for Type Script 2.7.1. The latest update of Visual Studio Code pushes Type Script 2.7.1 as the default, and I needed to revert back to the older version of Type Script just to keep using tslint.

**Comment by syq7970** [comment]
+1 I have same issue

**Comment by xaviergonz** [comment]
restarting vscode worked for me

**Comment by xzguan** [comment]
pay attention to the file tsconfig.json, comment the ("strict": true) in the compilerOptions file

**Comment by arpitparmar5739** [comment]
class Engine {
    constructor(public horsePower: number,
        public engineType: string) { }
}

class Car {
    private _engine: Engine;
    constructor(engine: Engine) {
        this.engine = engine;
    }

    get engine(): Engine {
        return this._engine;
    }

    set engine(engine: Engine) {
        if (engine === undefined) throw 'Supply Engine';
        this._engine = engine;
    }
}

let engine = new Engine(1000, "awesome");
let car = new Car(engine);

console.log(car.engine.engineType);

Error: Property '_engine' has no initializer and is not definitely assigned in the constructor.
I can turn off the strict property but can't i use it without turning off the strict mode!

**Comment by JoshuaKGoldberg** [comment]
Seems like these issues were all around moving to newer TypeScript and TSLint versions.

Quick tip for anyone finding this thread: you can check which version of TypeScript you're using in VS Code with the "Select TypeScript Version" command. 😉

![image](https://user-images.githubusercontent.com/3335181/47991951-9a791c00-e0b9-11e8-91bc-b9405fa72ea5.png)

![image](https://user-images.githubusercontent.com/3335181/47991939-90efb400-e0b9-11e8-9ae8-e9ec7213f011.png)
